### PR TITLE
Fix PacketOut validation

### LINKED ISSUE #402: packetOut validation needs fix
packetOut validation of in_port is not working correctly and needs fix so it can pass the tests properly.

### PR DESCRIPTION
**Pull request by macartur**
- Update packetout validation
- Fix tests
- Fix #402

### CODE REVIEW

File: pyof/v0x01/controller2switch/packet_out.py [hunk 1]
```diff
@@ -112,13 +112,14 @@ def _update_actions_len(self):
...
-            valid = False
-        if not valid:
-            raise ValidationError('{} is not a valid input port.'.format(port))
+        """Validate in_port attribute.
+
+        Raises:
+            ValidationError: If in_port is a invalid port.
+
+        """
+        is_valid_range = self.in_port > 0 and self.in_port <= Port.OFPP_MAX
```
Comment (diraol) [inline pyof/v0x01/controller2switch/packet_out.py]: @renanrodrigo, me and @macartur think that `Port.OFPP_MAX` should be a valid port value. Do you agree?

Comment (diraol) [inline pyof/v0x01/controller2switch/packet_out.py]: @macartur if `self.in_port` is an instance of `Port` (enum), then this comparison (`self.in_port > 0 and self.in_port <= Port.OFPP_MAX`) may not work.

I think that we have two cases here:

1. If `self.in_port` is an integer:
    - Then we should compare it to `0` (zero) and to `Port.OFPP_MAX.value`
2. If `self.in_port` is an instance of `Port`:
    - Then we don't need to do any check (or do we need to?)

File: pyof/v0x01/controller2switch/packet_out.py [hunk 2]
```diff
@@ -112,13 +112,14 @@ def _update_actions_len(self):
...
-        valid = True
-        if isinstance(port, Port):
-            if port not in _VIRT_IN_PORTS:
-                valid = False
-        elif isinstance(port, int) and (port < 1 or port >=
-                                        Port.OFPP_MAX.value):
-            valid = False
-        if not valid:
-            raise ValidationError('{} is not a valid input port.'.format(port))
+        """Validate in_port attribute.
```
Comment (diraol) [inline pyof/v0x01/controller2switch/packet_out.py]: @macartur Please, add a line here commenting the validation criteria. What is an `invalid port`?

File: pyof/v0x01/controller2switch/packet_out.py [hunk 3]
```diff
@@ -112,13 +112,14 @@ def _update_actions_len(self):
...
+
+        Raises:
+            ValidationError: If in_port is a invalid port.
```
Comment (diraol) [inline pyof/v0x01/controller2switch/packet_out.py]: **an** invalid port.

File: pyof/v0x01/controller2switch/packet_out.py (hunk 1 above)
Comment (renanrodrigo) [reply pyof/v0x01/controller2switch/packet_out.py]: Agreed. I do think it's valid.

Comment (macartur) [reply pyof/v0x01/controller2switch/packet_out.py]: @diraol this expression works well to me.
Using the ipython console I got:

Case 1:
```python
2 > 0 and 2 <= Port.OFPP_MAX
True
```
Case 2:
```python
655353 > 0 and 655353 <= Port.OFPP_MAX
False
```

Case 3:
```python
Port.OFPP_LOCAL > 0 and Port.OFPP_LOCAL <= Port.OFPP_MAX
False
```

Comment (renanrodrigo) [reply pyof/v0x01/controller2switch/packet_out.py]: @diraol , `Port` is an `IntEnum`, and thus its values can be compared to integers normally (:

Comment (diraol) [reply pyof/v0x01/controller2switch/packet_out.py]: Ok folks, thanks! I forgot about the `IntEnum` and was thinking about a `Enum` only. ;)

Comment (diraol) [reply pyof/v0x01/controller2switch/packet_out.py]: Tks!

**Pull request merged**